code/modules/mob/living/carbon/human/species_types/jellypeople.dm

```diff
@@ -479,88 +479,263 @@
 	linked_actions.Add(action)
-	action.Grant(M)
+	action.Grant(mind.current)
+	to_chat(mind.current, "<span class='notice'>You are now connected to [owner.real_name]'s Slime Link.</span>")
+	GLOB.slime_links_by_mind[mind] = WEAKREF(src)
+	register_mob_signals(mind.current, death = TRUE, mind_transfer = TRUE)
+	to_chat(mind.current, "<span class='big notice'>You can use :[MODE_KEY_SLIMELINK] or .[MODE_KEY_SLIMELINK] to talk over the slime link!</span>")
 	return TRUE
 
-/datum/species/oozeling/stargazer/proc/unlink_mob(mob/living/M)
-	var/link_id = linked_mobs.Find(M)
-	if(!(link_id))
+/**
+ * Unlinks a mind from this stargazer's slime link.
+ *
+ * Arguments
+ * * mind: The mind to unlink from the slime link.
+ */
+/datum/species/oozeling/stargazer/proc/unlink_mind(datum/mind/mind)
+	var/link_id = linked_minds.Find(mind)
+	if(!link_id)
 		return
-	var/datum/action/innate/linked_speech/action = linked_actions[link_id]
-	action.Remove(M)
 	var/mob/living/carbon/human/owner = slimelink_owner.resolve()
-	if(owner)
-		to_chat(M, "<span class='notice'>You are no longer connected to [owner.real_name]'s Slime Link.</span>")
-	linked_mobs -= M
+	var/datum/action/innate/linked_speech/action = linked_actions[link_id]
+	unregister_mob_signals(mind)
+	if(mind.current)
+		action.Remove(mind.current)
+		to_chat(mind.current, "<span class='notice'>You are no longer connected to [owner.real_name]'s Slime Link.</span>")
 	linked_actions -= action
-	qdel(action)
+	linked_minds -= mind
+	if(GLOB.slime_links_by_mind[mind])
+		var/datum/weakref/owner_ref = GLOB.slime_links_by_mind[mind]
+		if(owner_ref?.resolve() == src)
+			GLOB.slime_links_by_mind -= mind
+
+/**
+ * Sends a chat message over the slime link.
+ * Anything calling this proc should ensure to filter and sanitize the message beforehand!
+ *
+ * Arguments
+ * * user: The mob sending the message.
+ * * message: The message to send.
+ */
+/datum/species/oozeling/stargazer/proc/slime_chat(mob/living/user, message)
+	var/mob/living/carbon/human/link_owner = slimelink_owner.resolve()
+	if(!link_owner || !istype(link_owner))
+		return
+	if(!user || !user.mind)
+		return
+	if(!user == link_owner && !(user.mind in linked_minds))
+		to_chat(user, "<span class='warning'>The link seems to have been severed...</span>")
+		return
+	if(!length(message))
+		return
+	message = user.treat_message_min(trim(message, MAX_MESSAGE_LEN))
+	var/msg = "<i><font color=#008CA2>\[[link_owner.real_name]'s Slime Link\] <b>[user.real_name]:</b> [message]</font></i>"
+	log_directed_talk(user, link_owner, msg, LOG_SAY, "slime link")
+	var/list/targets = linked_minds.Copy()
+	if(link_owner.mind)
+		targets += link_owner.mind
+	for(var/datum/mind/linked_mind as anything in targets)
+		if(linked_mind != link_owner.mind && !validate_mind(linked_mind, silent = TRUE))
+			unlink_mind(linked_mind)
+			continue
+		to_chat(linked_mind.current, msg)
+
+	for(var/mob/target as anything in GLOB.dead_mob_list)
```

Comment:
why do you use `as anything` here. Do you really know what `as anything` or `as()` means?